code/game/objects/machinery/constructable_frame.dm

```diff
@@ -2,44 +2,89 @@
 	name = "machine frame"
 	icon = 'icons/obj/stock_parts.dmi'
 	icon_state = "box_0"
-	density = FALSE
-	anchored = TRUE
-	use_power = NO_POWER_USE
-	var/list/components
-	var/list/req_components
-	var/list/req_component_names
-	var/state = 1
-
-
-/obj/machinery/constructable_frame/proc/update_desc()
-	var/D
-	if(req_components)
-		D = "Requires "
-		var/first = 1
-		for(var/I in req_components)
-			if(req_components[I] > 0)
-				D += "[first?"":", "][num2text(req_components[I])] [req_component_names[I]]"
-				first = 0
-		if(first) // nothing needs to be added, then
-			D += "nothing"
-		D += "."
-	desc = D
-
-
-/obj/machinery/constructable_frame/state_2
-	icon_state = "box_1"
-	state = 2
-
-
-/obj/machinery/constructable_frame/machine_frame/attackby(obj/item/I, mob/living/user, params)
-	if(I.crit_fail)
-		to_chat(user, "<span class='warning'>This part is faulty, you cannot add this to the machine!</span>")
+	density = TRUE
+	max_integrity = 250
+	obj/item/circuitboard/machine/circuit = null
+	var/state = 0
+
+/obj/machinery/constructable_frame/examine(user)
+	. = ..()
+	if(circuit)
+		. += "It has \a [circuit] installed."
+
+
+/obj/machinery/constructable_frame/deconstruct(disassembled = TRUE)
```

Comment:
i dont know where you copy pasted this from but it is not compatible with our integrity system, look at tg's

https://github.com/tgstation/tgstation/blob/71e69cf7d80dd3e658099c4b78ec70555017ae14/code/game/machinery/constructable_frame.dm#L16

Reply:
Oh, I copied it directly from /tg/. Looks the same, I do not see the issue.

Reply:
well you didnt because tg uses `/obj/structure/frame` not `/obj/machinery/constructable_frame`

Reply:
Huh, you are correct. I probably changed it myself when I started it months ago since it has been a /structure/ object for years according to the history.